backend/test/richard_burton/publication_index_test.exs

```diff
@@ -0,0 +1,182 @@
+defmodule RichardBurton.Publication.IndexTest do
+  @moduledoc """
+  Tests for the Publication.Index module
+  """
+
+  use RichardBurton.DataCase
+
+  alias RichardBurton.Publication
+
+  setup(_context) do
+    {:ok, publications} = Publication.Codec.from_csv("test/fixtures/data_index.csv")
+
+    ps =
+      publications
+      |> Enum.map(&insert!/1)
+      |> Publication.preload()
+      |> Publication.Codec.flatten()
+      |> Enum.map(&Map.to_list/1)
+      |> Enum.map(&Enum.map(&1, fn {k, v} -> {String.to_existing_atom(k), v} end))
+      |> Enum.map(&Map.new/1)
+
+    [publications: ps]
+  end
+
+  defp insert!(attrs) do
+    %Publication{} |> Publication.changeset(attrs) |> Repo.insert!()
+  end
+
+  defp sort(publications) do
+    Enum.sort(publications)
+  end
+
+  defp filter(publications, key, term) do
+    Enum.filter(publications, &String.contains?(to_string(&1[key]), term))
+  end
+
+  describe "all/0" do
+    test "returns all publications", context do
```

Comment:
I would be natural to avoid the `context` binding..

```ex
test "returns all publications", %{publications: publications} do
```

Reply:
Makes sense, I did it like this to cleanup the test header a bit (and reduce the probabilities of the macro being split in two lines). But I guess it's a lost cause, that will happen eventually, and the binding is unnecessary.

I'll change it

Reply:
Superseded by the refactor on the assertions